### ISSUE TITLE
Tracker Fails to Record Changes Made to Deck in Tavern Encounter of Dalaran Heist WAS CLOSED PREMATURELY thank you.

### ISSUE THREAD

**Issue by Glitchex** [HearthSim/Hearthstone-Deck-Tracker#3917]
Here is my hdt_log.txt for when it happened tonight. Its alot to go through but I can see where a few problems are located.
[hdt_log_1559274832.txt](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/3239704/hdt_log_1559274832.txt)

_Originally posted by @Glitchex in https://github.com/HearthSim/Hearthstone-Deck-Tracker/issues/3915#issuecomment-497570137_

**Comment by MelTorefas** [comment]
Can confirm this is a problem. Seems to be related to Dismiss and/or Recruit. Round of Drinks works fine, as does The Gang's All Here. But after either Dismiss or Recruit, the deck tracker fails to update the adventure deck and instead creates a duplicate deck containing your deck's starting cards only. 

In game, it attempts to load the new deck, fails, and then loads the old deck, which does not contain the card added from Recruit and still shows the card removed by Dismiss. Once the Recruited card is drawn, the deck track freaks out because that card is 'not found', and usually switches to No Deck Mode.

I have included my log from right after finishing the next boss fight after the tavern, during which I drew the recruited card and the deck tracker switched to No Deck Mode.
[hdt_log.txt](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/3242586/hdt_log.txt)

**Comment by Glitchex** [comment]
> Can confirm this is a problem. Seems to be related to Dismiss and/or Recruit. Round of Drinks works fine, as does The Gang's All Here. But after either Dismiss or Recruit, the deck tracker fails to update the adventure deck and instead creates a duplicate deck containing your deck's starting cards only.
> 
> In game, it attempts to load the new deck, fails, and then loads the old deck, which does not contain the card added from Recruit and still shows the card removed by Dismiss. Once the Recruited card is drawn, the deck track freaks out because that card is 'not found', and usually switches to No Deck Mode.
> 
> I have included my log from right after finishing the next boss fight after the tavern, during which I drew the recruited card and the deck tracker switched to No Deck Mode.
> [hdt_log.txt](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/3242586/hdt_log.txt)

**Comment by Glitchex** [comment]
I did not mean to close this issue, as I thought it meant delete my comment... so sorry about that, Im not hip to how this forum works.

**Comment by Glitchex** [comment]
Thank you MelTorefas for figuring this out. Hopefully they will patch or whatever.

**Comment by DustinCon** [comment]
And for some reason if you try and manually add or remove the cards tavern changed, it pushes the deck to wild instead of leaving it in dungeon. So when I try and use the deck I made, it freaks out and auto-makes the original deck from the start of the run and than freaks out cause all the cards are wrong.

**Comment by MelTorefas** [comment]
On further playthroughs this issue seems to be tied to removing cards, not adding them. I can use any of the cards that add things to the deck (Recruit, Round of Drinks, The Gang's All Here, and the one that discovers a legendary minion) and it will be fine. But any of the removal effects (Dismiss, You're All Fired, and Kindle) seem to cause it to fail to update and then break for the rest of the run.

**Comment by Snowpup** [comment]
Just confirming this issue affects me as well, persisting to this day.

**Comment by lindydonna** [comment]
I'm seeing the same issue. I can't confirm if it's related only to actions that remove cards from the deck, but I am seeing that when a run nears the end, the deck tracker has no info on the deck.

**Comment by DustinCon** [comment]
Just popping in to say the issue still seems to be persisting. At least for me.

**Comment by ivivaitylin** [comment]
2 months on, issue still exists.

**Comment by asakurato** [comment]
yeah, had this issue since Dalaran and it seems that Recruit or dismiss causes that, but possibly mana reductions too.
